### PR TITLE
FIX-#4347: read_excel: defaults to pandas for unsupported types of 'io'

### LINKED ISSUE #4347: `read_excel` raises exception when passing in-memory file bytes object
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`): 0.13.3
- **Python version**: 3.8.2
- **openpyxl version**: 3.0.9

Current implementation doesn't allow passing `bytes` or `BytesIO` object, unlike standard pandas implementation:
```
import modin.pandas as pd

xlsx_file = pd.ExcelFile(file_content) # passing bytes or BytesIO object
pd.read_excel(xlsx_file)
```
Output:
```
File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/modin/pandas/io.py", line 353, in read_excel
    intermediate = FactoryDispatcher.read_excel(**kwargs)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/modin/core/execution/dispatching/factories/dispatcher.py", line 217, in read_excel
    return cls.__factory._read_excel(**kwargs)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/modin/core/execution/dispatching/factories/factories.py", line 257, in _read_excel
    return cls.io_cls.read_excel(**kwargs)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/modin/core/io/file_dispatcher.py", line 151, in read
    query_compiler = cls._read(*args, **kwargs)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/modin/core/io/text/excel_dispatcher.py", line 84, in _read
    ex = ExcelReader(io_file, read_only=True)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/openpyxl/reader/excel.py", line 124, in __init__
    self.archive = _validate_archive(fn)
  File "/Users/dchernov/Projects/test/.venv/lib/python3.8/site-packages/openpyxl/reader/excel.py", line 78, in _validate_archive
    file_format = os.path.splitext(filename)[-1].lower()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/posixpath.py", line 118, in splitext
    p = os.fspath(p)
TypeError: expected ExcelFile.__fspath__() to return str or bytes, not _io.BytesIO
```

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4347 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/io/io.py [hunk 1]
```diff
@@ -277,6 +278,8 @@ def read_clipboard(cls, sep=r"\s+", **kwargs):  # pragma: no cover # noqa: PR01
     )
     def read_excel(cls, **kwargs):  # noqa: PR01
         ErrorMessage.default_to_pandas("`read_excel`")
+        if isinstance(kwargs["io"], ExcelFile):
+            kwargs["io"]._set_pandas_mode()
```
Comment (vnlitvinov) [inline modin/core/io/io.py:282]: this looks super-weird, could you please add an in-code comment explaining the _why this is needed_?

File: modin/pandas/io.py [hunk 2]
```diff
@@ -835,10 +835,19 @@ class ExcelFile(ClassLogger, pandas.ExcelFile):  # noqa: PR01, D200
     Class for parsing tabular excel sheets into DataFrame objects.
     """
 
+    _behave_like_pandas = False
+
+    def _set_pandas_mode(self):  # noqa
+        # disable Modin behavior to be able to pass object to `pandas.read_excel`
+        self._behave_like_pandas = True
```
Comment (vnlitvinov) [inline modin/pandas/io.py:842]: this feels like a hack... would like to see an explanation of the flow

File: modin/pandas/test/test_io.py [hunk 3]
```diff
@@ -2090,6 +2091,51 @@ def test_ExcelFile(self, make_excel_file):
...
+    def test_ExcelFile_bytes(self, make_excel_file):
+        unique_filename = make_excel_file()
+        with open(unique_filename, mode="rb") as f:
+            content = f.read()
+
+        modin_excel_file = pd.ExcelFile(content)
+        pandas_excel_file = pandas.ExcelFile(content)
+
+        df_equals(modin_excel_file.parse(), pandas_excel_file.parse())
+        assert isinstance(modin_excel_file, pd.ExcelFile)
```
Comment (vnlitvinov) [inline modin/pandas/test/test_io.py:2103]: what is the point of this check?.. shouldn't it be `isinstance(modin_excel_file, pandas.ExcelFile)` or whatever?

Comment (anmyachev) [reply modin/pandas/test/test_io.py]: Removed

**Comment by anmyachev** [comment]
@vnlitvinov ready for review

**Review by vnlitvinov — APPROVED** [review]
LGTM!

File: modin/core/io/text/excel_dispatcher.py [hunk 4]
```diff
@@ -55,6 +59,22 @@ def _read(cls, io, **kwargs):
                 **kwargs
             )
 
+        if isinstance(io, bytes):
+            io = BytesIO(io)
+
+        # isinstance(ExcelFile, os.PathLike) == True
```
Comment (YarShev) [inline modin/core/io/text/excel_dispatcher.py:65]: Was this left intentionaly?

Comment (anmyachev) [reply modin/core/io/text/excel_dispatcher.py:65]: Yes

**Pull request merged**